DataFormats/MuonReco/interface/MuonEnergy.h

```diff
@@ -55,32 +64,37 @@ namespace reco {
        /// RecHits
        ///
        /// crossed HCAL rechits
-       const std::vector<HBHERecHit>* getHadRecHits() const
+       std::vector<HcalMuonRecHit> crossedHadRecHits;
+       bool crossedHadRecHitsValid;
+
+       const std::vector<HcalMuonRecHit>* getCrossedHadRecHits() const
        {
-	 if (hadRecHitsValid) return &hadRecHits;
+	 if (crossedHadRecHitsValid) return &crossedHadRecHits;
 	 return nullptr;
        }
 		 
-       void setHadRecHits(const std::vector<const HBHERecHit*>& rechits)
+       void setCrossedHadRecHits(const std::vector<const HBHERecHit*>& rechits)

```

Comment:
it would be nice to keep this struct as simple as possible (following its original design of a simple struct)
and avoid dependence on another complex class (HBHERecHit). 
This modifier assignment loop can be implemented in the producer.

I'm somewhat puzzled why this code compiles (or at least passes the header consistency checks) because there is no dependency in  DataFormats/MuonReco BuildFile on DataFormats/HcalRecHit.

Reply:
ok